### ISSUE TITLE
GitHub の使い方に関する検討事項

### ISSUE THREAD

**Issue by yofukami** [icat-lab/g4n#3]
# 概要 
[GitHub の使い方](https://github.com/icat-lab/g4n/blob/main/howto/github.md)に関する議論の場です。いきなりpull requestよりは、と思いまして。

# 検討事項
## 学習院基礎ゼミ向けバージョンの作成検討
学習院基礎ゼミ向けに、別リポジトリで作成する必要があると思います。というのも、初学者、GitHubに初めて触れる学生がほとんどですので、ある程度彼らのリテラシーと前提条件を反映した表記にするほうが、スムーズに入れるだろうと思われるためです。
基礎ゼミのリポジトリの、[ここのディレクトリ](https://github.com/icat-lab/icat_lab/tree/master/basic_seminar_undergrad)に、基礎ゼミバージョンを作る or 1学期/2学期の学生の作業用リポジトリが良いかと思います。
https://github.com/icat-lab/icat_lab/tree/master/basic_seminar_undergrad

### [アカウントの作成](https://github.com/icat-lab/g4n/blob/yoshiaki/howto/github.md#%E3%82%A2%E3%82%AB%E3%82%A6%E3%83%B3%E3%83%88%E3%81%AE%E4%BD%9C%E6%88%90)
1. Username は GitHub 内で使われていない名前
「Username は GitHub 内で使われていない名前」って、同一リポジトリ内で使われていないってことですか？

2. パソコンで参照可能なメールアドレス
「パソコンで参照可能なメールアドレス」というのは、パソコンで送受信できる、という意味ですか？

1. 組織のメールアドレス
「組織のメールアドレスよりは個人のメールアドレスにしておいた方が無難でしょう」ですが、学生の場合「組織」というのは、　「gakushuin.ac.jpドメインのアドレス」と指定しないとわかりにくいですね。学生向けの授業の場合はある程度決め打ちで指示するほうが混乱しなかったりするため、最初からGmail指定、なんてことを2020年度ではやりましたけど、そこまでするかは要検討です。

**Comment by Gucchiy** [comment]
@yofukami 

> 1. Username は GitHub 内で使われていない名前
> 「Username は GitHub 内で使われていない名前」って、同一リポジトリ内で使われていないってことですか？

いいえ、どうも GitHub 全体でユニークである必要があるようです。Twitter アカウントみたいなものらしいのですが、「GitHub 内で使われていない」では分からないとすると、どう表現したらよいかですが、、他の人が使っている Username だと、その場でリアルタイムにエラー表示になるので「入れてみれば分かる」ような気もしています。

> 2. パソコンで参照可能なメールアドレス
> 「パソコンで参照可能なメールアドレス」というのは、パソコンで送受信できる、という意味ですか？

おっしゃる通りです。事業者メールでスマホ上でしか使えない～というところが回避したかったところです。POP メーラーでも Web メールでもいいから、パソコンで見えるようにしてね、と言うメッセージでした。もちろん、多くの事業者メールはパソコン上でも送受信できるわけですが、それが分かる人はそれでもいいよ、というニュアンスでこの表現にしています。分かりにくいでしょうかね・・

> 3. 組織のメールアドレス
> 「組織のメールアドレスよりは個人のメールアドレスにしておいた方が無難でしょう」ですが、学生の場合「組織」というのは、　「gakushuin.ac.jpドメインのアドレス」と指定しないとわかりにくいですね。学生向けの授業の場合はある程度決め打ちで指示するほうが混乱しなかったりするため、最初からGmail指定、なんてことを2020年度ではやりましたけど、そこまでするかは要検討です。

了解です、とりあえずは「gakushuin.ac.jpドメインのアドレス」 に修正しましょう。

**Comment by yofukami** [comment]
> 基礎ゼミのリポジトリの、ここのディレクトリに、基礎ゼミバージョンを作る or 1学期/2学期の学生の作業用リポジトリが良いかと思います。

issuesを活用してゼミを行うのであれば、学期ごとにリポジトリを立てるのが適切だと思います。
@Gucchiy  2021/3/19のteleconfで確認したいです。

**Comment by yofukami** [comment]
1学期のリポジトリを立てました。
[2021年度1学期経営入門演習(深見ゼミ)作業スペース](https://github.com/icat-lab/ay2021spring_icatlab)

**Comment by Gucchiy** [comment]
決まった要件は、 #8 に書くようにしてみました。ここではディスカッションと進捗を記載します。

**Comment by Gucchiy** [comment]
さっそくですが、ファイル構成を下記通り修正です。[文書の Commit ログ](https://github.com/icat-lab/g4n/commit/bc7a30e586b63d5c4fa058b03159f22d8dba7f4a)を活用するのが本質ですが、まずは慣れるまでこれで・・
- [github.md トップ ページ](https://github.com/icat-lab/g4n/blob/main/howto/github.md)
- [github_login.md アカウントの作成](https://github.com/icat-lab/g4n/blob/main/howto/github_login.md)
- [github_issues.md Issues を使う](https://github.com/icat-lab/g4n/blob/main/howto/github_issues.md)
- [github_projects.md Projects を使う](https://github.com/icat-lab/g4n/blob/main/howto/github_projects.md)

**Comment by yofukami** [comment]
# 概要
第2回ゼミ後のラップアップで出た話をここにメモしておきます。

# GUIでどこまでできる？
- 少なくとも1学期はGUIベースで完結させる前提で、もしできるようであればCUIに踏み込むかも、的なスタンスで。
- グループ代表者のcodeへの投稿は、GUI上でできるのか、要確認
- 代表者の投稿に対する、他メンバーからのpullu requestがGUIでできるのか、要確認。

**Comment by Gucchiy** [comment]
@yofukami 
code への投稿用インストラクション「リポジトリを使う」を追加しました！
https://github.com/icat-lab/g4n/blob/main/howto/github_repository.md
学生へのご指示に適宜ご活用くださいませ。あれ？これ私が学生に直接言って良いんでしたっけ？

**Comment by yofukami** [comment]
@Gucchiy 返信が遅くなり失礼いたしました。これ、直接言っていただいて結構です! 恐らく、GitHubの操作に関することは @Gucchiy が話されたほうが、担当教員以外の方が関わっている根拠が強くなりますので.

**Comment by yofukami** [comment]
ちなみに、今fileの修正を直接GUIでやってみました。私がリポジトリのオーナーだからできるって話でありますよね？
https://github.com/icat-lab/g4n/commit/77106ae94b511c5bdb6f7505a42b5a78ef2e45f0

**Comment by Gucchiy** [comment]
@yofukami あ、確かにそれはありますね！ちょっとテストしてみましょう。プルリクしか許されないかも。ちなみに、どのフォルダに課題を配置してもらいますか？

**Comment by Gucchiy** [comment]
@yofukami Read ユーザーだと、プルリクになることが分かりましたー。しかし、学生は Write ユーザー（管理権限はないが、push は許されている）なので、今の説明で大丈夫です
